### PR TITLE
Add configuration & usage documentation

### PR DESCRIPTION
**Pull request by lguychard**
We will likely link to these docs from the Atlassian Marketplace listing, and users will be able to access them directly from the admin panel after installing the plugin:

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,39 @@
-# Sourcegraph Bitbucket Server Integration
+# Sourcegraph for Bitbucket
```
Comment (christinaforney) [inline README.md]: I think we do need it to be Bitbucket Server (since Bitbucket Cloud is an entirely different thing)

File: pom.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <version>1.0.0</version>
     <packaging>atlassian-plugin</packaging>
 
-    <name>Sourcegraph</name>
+    <name>Sourcegraph for Bitbucket</name>
```
Comment (christinaforney) [inline pom.xml]:
```suggestion
    <name>Sourcegraph for Bitbucket Server</name>
```

**Review by chrismwendt — APPROVED** [review]
Looks great :+1:

File: README.md [hunk 3]
```diff
@@ -1,6 +1,39 @@
...
+
+![Add-ons](img/add-ons.png)
+
+3. On the **Sourcegraph Settings** page, set the Sourcegraph URL to the URL of your self-hosted Sourcegraph instance:
+
+![Sourcegraph settings](img/sourcegraph-settings.png)
+
+### Usage
+
+Once configured, Sourcegraph for Bitbucket will add code intelligence hovers to code views and pull requests for all users that are logged in to your self-hosted Sourcegraph instance:
```
Comment (christinaforney) [inline README.md]: Perhaps mention what would happen if a user is not logged in to the Sourcegraph instance to pre-empt the question for admins choosing to set up for their company.

**Review by christinaforney — APPROVED** [review]

**Pull request merged**